Terminal.Gui/ConsoleDrivers/WindowsDriver.cs

```diff
@@ -1853,39 +1851,44 @@ bool IMainLoopDriver.EventsPending (bool wait)
 		}
 
 		try {
-			if (!_tokenSource.IsCancellationRequested) {
-				_eventReady.Wait (waitTimeout, _tokenSource.Token);
+			if (!_eventReadyTokenSource.IsCancellationRequested) {
+				_eventReady.Wait (waitTimeout, _eventReadyTokenSource.Token);
 			}
 		} catch (OperationCanceledException) {
 			return true;
 		} finally {
 			_eventReady.Reset ();
 		}
 
-		if (!_tokenSource.IsCancellationRequested) {
+		if (!_eventReadyTokenSource.IsCancellationRequested) {
 			return _resultQueue.Count > 0 || CheckTimers (wait, out _) || _winChanged;
 		}
 
-		_tokenSource.Dispose ();
-		_tokenSource = new CancellationTokenSource ();
+		_eventReadyTokenSource.Dispose ();
+		_eventReadyTokenSource = new CancellationTokenSource ();
 		return true;
 	}
-
+	
 	bool CheckTimers (bool wait, out int waitTimeout)
```

Comment:
This is a common code used by all mainloop. Why not make it a public method on the `MainLoop` class? Only is needed to add an additional parameter `IMainLoop` to pass the `_mainLoop` field.

Reply:
Good suggestion. Let's do this in #2872